### PR TITLE
Improved chapters 'Implicit Coercions' and 'Canonical Structures' of the Reference Manual.

### PR DESCRIPTION
**Pull request by wkolowski**
**Kind:** documentation

Mostly grammar and spelling errors.

Note that in the chapter about canonical structures there are some warnings in the examples and not a single word is written to explain them.

### CODE REVIEW

File: doc/sphinx/addendum/canonical-structures.rst [hunk 1]
```diff
@@ -396,7 +396,7 @@ instances of the ``LEQ`` class.
 
   Definition packager T e0 le0 (m0 : LEQ.mixin e0 le0) :=
     [find e | EQ.obj e ~ T | "is not an EQ.type" ]
-    [find o | LE.obj o ~ T | "is not an LE.type" ]
+    [find o | LE.obj o ~ T | "is not a LE.type" ]
```
Comment (anton-trunov) [inline doc/sphinx/addendum/canonical-structures.rst]: I pronounce `LE` as `el-iː`, so it seems that `an` would be correct the correct option, unless there is some language rule I don't know about.

File: doc/sphinx/addendum/implicit-coercions.rst [hunk 2]
```diff
@@ -121,7 +121,7 @@ by the coercions ``f₁..fₖ``.  The application of a coercion path to a
 term consists of the successive application of its coercions.
 
 
-Declaration of Coercions
+Declaring of Coercions
```
Comment (anton-trunov) [inline doc/sphinx/addendum/implicit-coercions.rst]: It seems to me that `of` is not needed here: "Declaring Coercions"

**Comment by gares** [comment]
@maximedenes Can't we suppress these warnings? I don't see the usual warning name...

  https://coq.inria.fr/distrib/current/refman/addendum/canonical-structures.html?highlight=Warning

File: doc/sphinx/addendum/implicit-coercions.rst (hunk 2 above)
Comment (Zeimer) [reply doc/sphinx/addendum/implicit-coercions.rst]: Right, trying to make it sound better I introduced even more errors : )

**Review by gares — APPROVED** [review]

**Review by Zimmi48 — APPROVED** [review]

**Pull request merged**